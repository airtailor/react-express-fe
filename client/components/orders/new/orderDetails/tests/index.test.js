import React from 'react';
import { shallow, mount, render } from 'enzyme';
import { OrderDetails } from '../';

const props = {
  updateCartCustomerInfo: jest.fn(),
  updateCartShipTo: jest.fn(),
  renderStageOne: jest.fn(),
  cart: {
    garments: [],
    customerInfo: {
      first_name: '',
      last_name: '',
      phone: '',
      email: '',
      street: '',
      street_two: '',
      city: '',
      state_province: '',
      zip_code: '',
      agrees_to_terms: true,
    },
    storeInfo: {},
    shipToStore: true,
    notes: '',
  },
};

describe('<OrderDeails />', () => {
  it('should render correctly', () => {
    const output = shallow(<OrderDetails {...props} />);
    expect(output).toMatchSnapshot();
  });
});
