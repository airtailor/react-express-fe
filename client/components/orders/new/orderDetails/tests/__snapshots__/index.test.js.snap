// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderDeails /> should render correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="order-details"
  >
    <h2>
      ORDER DETAILS
    </h2>
    <CustomerInfo
      customerInfo={
        Object {
          "agrees_to_terms": true,
          "city": "",
          "email": "",
          "first_name": "",
          "last_name": "",
          "phone": "",
          "state_province": "",
          "street": "",
          "street_two": "",
          "zip_code": "",
        }
      }
      updateCustomerInfo={[Function]}
    />
    <h3>
      Shipping
    </h3>
    <div>
      <br />
      <div>
        <Checkbox
          checked={true}
          name="ship-to-store"
          onChange={[Function]}
          text="Ship To Store"
        />
        <br />
        <br />
        <Checkbox
          checked={false}
          name="ship-to-customer"
          onChange={[Function]}
          text="Ship To Customer"
        />
        <br />
        <br />
      </div>
    </div>
  </div>,
  "nodes": Array [
    <div
      className="order-details"
    >
      <h2>
        ORDER DETAILS
      </h2>
      <CustomerInfo
        customerInfo={
          Object {
            "agrees_to_terms": true,
            "city": "",
            "email": "",
            "first_name": "",
            "last_name": "",
            "phone": "",
            "state_province": "",
            "street": "",
            "street_two": "",
            "zip_code": "",
          }
        }
        updateCustomerInfo={[Function]}
      />
      <h3>
        Shipping
      </h3>
      <div>
        <br />
        <div>
          <Checkbox
            checked={true}
            name="ship-to-store"
            onChange={[Function]}
            text="Ship To Store"
          />
          <br />
          <br />
          <Checkbox
            checked={false}
            name="ship-to-customer"
            onChange={[Function]}
            text="Ship To Customer"
          />
          <br />
          <br />
        </div>
      </div>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <OrderDetails
      cart={
        Object {
          "customerInfo": Object {
            "agrees_to_terms": true,
            "city": "",
            "email": "",
            "first_name": "",
            "last_name": "",
            "phone": "",
            "state_province": "",
            "street": "",
            "street_two": "",
            "zip_code": "",
          },
          "garments": Array [],
          "notes": "",
          "shipToStore": true,
          "storeInfo": Object {},
        }
      }
      renderStageOne={[Function]}
      updateCartCustomerInfo={[Function]}
      updateCartShipTo={[Function]}
    />,
    "_forcedUpdate": false,
    "_instance": OrderDetails {
      "context": Object {},
      "props": Object {
        "cart": Object {
          "customerInfo": Object {
            "agrees_to_terms": true,
            "city": "",
            "email": "",
            "first_name": "",
            "last_name": "",
            "phone": "",
            "state_province": "",
            "street": "",
            "street_two": "",
            "zip_code": "",
          },
          "garments": Array [],
          "notes": "",
          "shipToStore": true,
          "storeInfo": Object {},
        },
        "renderStageOne": [Function],
        "updateCartCustomerInfo": [Function],
        "updateCartShipTo": [Function],
      },
      "refs": Object {},
      "state": Object {},
      "updateCustomerInfo": [Function],
      "updater": Updater {
        "_renderer": [Circular],
      },
    },
    "_newState": null,
    "_rendered": <div
      className="order-details"
    >
      <h2>
        ORDER DETAILS
      </h2>
      <CustomerInfo
        customerInfo={
          Object {
            "agrees_to_terms": true,
            "city": "",
            "email": "",
            "first_name": "",
            "last_name": "",
            "phone": "",
            "state_province": "",
            "street": "",
            "street_two": "",
            "zip_code": "",
          }
        }
        updateCustomerInfo={[Function]}
      />
      <h3>
        Shipping
      </h3>
      <div>
        <br />
        <div>
          <Checkbox
            checked={true}
            name="ship-to-store"
            onChange={[Function]}
            text="Ship To Store"
          />
          <br />
          <br />
          <Checkbox
            checked={false}
            name="ship-to-customer"
            onChange={[Function]}
            text="Ship To Customer"
          />
          <br />
          <br />
        </div>
      </div>
    </div>,
    "_rendering": false,
    "_updater": Updater {
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <OrderDetails
    cart={
      Object {
        "customerInfo": Object {
          "agrees_to_terms": true,
          "city": "",
          "email": "",
          "first_name": "",
          "last_name": "",
          "phone": "",
          "state_province": "",
          "street": "",
          "street_two": "",
          "zip_code": "",
        },
        "garments": Array [],
        "notes": "",
        "shipToStore": true,
        "storeInfo": Object {},
      }
    }
    renderStageOne={[Function]}
    updateCartCustomerInfo={[Function]}
    updateCartShipTo={[Function]}
  />,
}
`;
